core/templates/dev/head/pages/library/Library.js

```diff
@@ -80,6 +80,41 @@ oppia.controller('Library', [
       $http.get('/libraryindexhandler').success(function(data) {
         $scope.libraryGroups = data.activity_summary_dicts_by_category;
 
+        $scope.userIsLoggedIn = GLOBALS.userIsLoggedIn;
+        var activityDetails = {};
+        if ($scope.userIsLoggedIn) {
+          $scope.activityPlaylistData = {explorations: {}, collections: {}};
+          $http.get('/creatordashboardhandler/data')
+            .then(function(response) {
+              $scope.libraryGroups.forEach(function(activityGroups) {
+                var activity = activityGroups.activity_summary_dicts;
+
+                activity.forEach(function(activity) {
+                  response.data.explorations_list.forEach(
+                    function(responseExploration) {
+                      if (responseExploration.id === activity.id) {
+                        $scope.activityPlaylistData
```

Comment:
Something is conceptually wrong here. You're getting data from /creatordashboard, which IIRC has no notion of playlists. The names may be off.

Reply:
I'm getting the list of activities that are created by the logging user. If there is any user who has authored the activity, it won't be available to add to playlist to the user as learner. This gets the work done in just one http call. Shall I add comments here for future developers?

Reply:
I think the variable names are wrong. They should describe the stuff coming from the backend, which doesn't have anything to do with "activity playlist data".

What exactly is the definition of the data returned from /creatordashboard? What does it represent?

Reply:
How about changing ```responseExploration/reponseCollection``` to ```userExplorationData/userCollectionData``` and ```activityPlaylistData``` to ```userActivityOwnership```?

Reply:
Better, but "data" is a bit vague. Perhaps responseExploration/responseCollection --> ownedExploration/ownedCollection (if the /creatordashboard response returns stuff that was created by the logged-in user)?

And then maybe activityPlaylistData --> activitiesOwned

Reply:
Sounds good to me.